src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,130 @@
+<?php
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver;
+
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+
+final class OptionConfigurator
+{
+    /**
+     * The name of the defined option
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var OptionsResolver
+     */
+    private $resolver;
+
+    public function __construct(string $name, OptionsResolver $resolver)
+    {
+        $this->name = $name;
+        $this->resolver = $resolver;
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed types for this option.
+     *
+     * @param string|string[] $types One or more accepted types
+     *
+     * @return $this
+     *
+     * @throws UndefinedOptionsException If the option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
+    public function allowedTypes($types): self
+    {
+        $this->resolver->addAllowedTypes($this->name, $types);
+
+        return $this;
+    }
+
+    /**
+     * Sets allowed values for this option.
+     *
+     * @param mixed $values One or more acceptable values/closures
+     *
+     * @return $this
+     *
+     * @throws UndefinedOptionsException If the option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
+    public function allowedValues($values): self
+    {
+        $this->resolver->setAllowedValues($this->name, $values);
+
+        return $this;
+    }
+
+    /**
+     * Sets the default value of a given option.
+     *
+     * @param mixed $value The default value of the option
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function default($value): self
+    {
+        $this->resolver->setDefault($this->name, $value);
+
+        return $this;
+    }
+
+    /**
+     * Marks this option as deprecated.
+     *
+     * @return $this
+     */
+    public function deprecated(): self
```

Comment:
this is missing support for custom deprecation messages and conditional deprecations (which both rely on the second argument of `setDeprecated`)